tests/i18n/test_extraction.py

```diff
@@ -127,6 +127,33 @@ def assertNotRecentlyModified(self, path):
         self.assertGreater(delta, 10, "%s wasn't recently modified" % path)
 
 
+class NormalizeLocaleTest(ExtractorTests):
+
+    def test_makemessages_with_normalized_locale(self):
+        """
+        Test the makemessages works with normalized locales in the input.
+        """
+
+        for locale in ('zh-cn', 'zh_cn', 'zh_CN', 'ZH-CN',):
+            out = StringIO()
+
+            with self.subTest():
```

Comment:
You should pass `locale` into `self.subTest()` otherwise there is no benefit - the whole point is that it will tell you which value failed the assertions in the scope of the with statement..